### ISSUE TITLE
Phangorn installation issue on R 3.5.3+ (incl. 3.6.0)

### ISSUE THREAD

**Issue by jananiravi** [KlausVigo/phangorn#72]
Hello,
I am trying to install `phangorn` which depends on `quadprog` for the latest versions of R (3.5.3 or 3.6.0). I am not able to install these packages without errors. 

### Temporary fix??
I finally went to the respective repositories and downloaded the `.tgz` files and forced them into the `R/3.5/library` folder on my mac. `Quadprog` now loads fine but `phangorn` throws the following error:
```
Error: package or namespace load failed for ‘phangorn’ in rbind(info, getNamespaceInfo(env, "S3methods")):
 number of columns of matrices must match (see arg 2)
In addition: Warning message:
package ‘phangorn’ was built under R version 3.6.0
```

### Normal installation still doesn't work
last few lines of error are as follows:
```
clang++ -std=gnu++11 -dynamiclib -Wl,-headerpad_max_install_names -undefined dynamic_lookup -single_module -multiply_defined suppress -L/Library/Frameworks/R.framework/Resources/lib -L/usr/local/lib -o phangorn.so RcppExports.o dist.o fitch.o init.o ml.o phangorn.o phangorn_help.o sankoff.o sprdist.o -L/Library/Frameworks/R.framework/Resources/lib -lRlapack -L/Library/Frameworks/R.framework/Resources/lib -lRblas -L/usr/local/gfortran/lib/gcc/x86_64-apple-darwin15/6.1.0 -L/usr/local/gfortran/lib -lgfortran -lquadmath -lm -F/Library/Frameworks/R.framework/.. -framework R -Wl,-framework -Wl,CoreFoundation
ld: warning: directory not found for option '-L/usr/local/gfortran/lib/gcc/x86_64-apple-darwin15/6.1.0'
ld: warning: directory not found for option '-L/usr/local/gfortran/lib'
ld: library not found for -lgfortran
clang: error: linker command failed with exit code 1 (use -v to see invocation)
make: *** [phangorn.so] Error 1
ERROR: compilation failed for package ‘phangorn’
* removing ‘/Users/janani/Library/R/3.5/library/phangorn’
* restoring previous ‘/Users/janani/Library/R/3.5/library/phangorn’
Warning in install.packages :
  installation of package ‘phangorn’ had non-zero exit status

The downloaded source packages are in
	‘/private/var/folders/t_/y5j0km_97w1b40s7prfpn6_00000gn/T/RtmpsllYmg/downloaded_packages’
```

**Tips? Suggestions?**
Any suggestions on the best way to install these two without encountering errors?
The phylogeny packages `msa` and `phytools` depend on `phangorn`. It would be great to get a fix. thanks!

***

### My environment
```
R version 3.5.3 Patched (2019-03-11 r76439)
Platform: x86_64-apple-darwin15.6.0 (64-bit)
Running under: macOS Mojave 10.14.4

Matrix products: default
BLAS: /System/Library/Frameworks/Accelerate.framework/Versions/A/Frameworks/vecLib.framework/Versions/A/libBLAS.dylib
LAPACK: /Library/Frameworks/R.framework/Versions/3.5/Resources/lib/libRlapack.dylib

locale:
[1] en_US.UTF-8/en_US.UTF-8/en_US.UTF-8/C/en_US.UTF-8/en_US.UTF-8

attached base packages:
[1] stats     graphics  grDevices utils     datasets  methods   base     

other attached packages:
[1] quadprog_1.5-5 ape_5.3       

loaded via a namespace (and not attached):
 [1] Rcpp_1.0.1         lattice_0.20-38    packrat_0.5.0-5    bitops_1.0-6      
 [5] grid_3.5.3         jsonlite_1.6       nlme_3.1-137       magrittr_1.5      
 [9] Matrix_1.2-15      fastmatch_1.1-0    tools_3.5.3        igraph_1.2.4.1    
[13] RCurl_1.95-4.12    yaml_2.2.0         parallel_3.5.3     rsconnect_0.8.13  
[17] compiler_3.5.3     pkgconfig_2.0.2    askpass_1.1        BiocManager_1.30.4
[21] openssl_1.3  
```

**Versions:**
```
               _                                          
platform       x86_64-apple-darwin15.6.0                  
arch           x86_64                                     
os             darwin15.6.0                               
system         x86_64, darwin15.6.0                       
status         Patched                                    
major          3                                          
minor          5.3                                        
year           2019                                       
month          03                                         
day            11                                         
svn rev        76439                                      
language       R                                          
version.string R version 3.5.3 Patched (2019-03-11 r76439)
nickname       Great Truth 
```

**Comment by KlausVigo** [comment]
Hi @jananiravi, 
I am not using OS X, but to compile `quadprog` and `phangorn` yourself,  you need Xcode (https://developer.apple.com/xcode/) and gfortran (https://gcc.gnu.org/wiki/GFortranBinaries) installed on your mac. It looks like you are missing gfortran (`ld: library not found for -lgfortran`).  With this you should also be able to install packages from github with `remotes::install_github` which contain C/C++ or FORTRAN code.  
Regards, 
Klaus

**Comment by jananiravi** [comment]
Thanks for your response, @KlausVigo. I have gcc, gfortran, and x-code up-to-date.
The force installing of `phangorn` and `quadprog` helped. Is there support/is there a way to install `quadprog` and `phangorn` directly in R v3.5.3 or v3.6 without these errors?
Thank you again!

**Comment by KlausVigo** [comment]
Hi @jananiravi , I hope this issue resolves itself when R 3.6.0. and quadprog 1.5.6. are both up to date on the CRAN servers, both have been updated recently (and need a specific version of each other). That seems to cause problems sometimes, but often after a few days these problems vanish. I keep an eye on it.

**Comment by jananiravi** [comment]
Thanks, @KlausVigo. Once quadprog and the other Bioconductor packages have been updated, I can comfortably move to `R v3.6`.
